core/arch/arm/include/kernel/generic_boot.h

```diff
@@ -31,10 +31,12 @@
 
 #if defined(CFG_WITH_ARM_TRUSTED_FW)
 uint32_t cpu_on_handler(uint32_t a0, uint32_t a1);
-uint32_t *generic_boot_init_primary(uint32_t pageable_part);
+uint32_t *generic_boot_init_primary(uint32_t pageable_part, uint32_t unused,
+				    uint32_t fdt);
```

Comment:
I imagine that if we ever start to map addresses >4GB on an LPI32 system we would need to change the definition of `paddr_t` for at least those systems. So far we haven't had to address that though. Until it becomes a real issue I think we should continue to require the physical pointers passed to OP-TEE at init to be <4GB.

On a LP64 system on the other hand, `unsigned long` will become a 64-bit type so we'd be able to use >4GB physical addresses there.